frontend2/src/components/compete/chart/chartUtils.ts

```diff
@@ -0,0 +1,81 @@
+import type { HistoricalRating, Tournament } from "api/_autogen";
+import type Highcharts from "highcharts";
+
+type UTCMilliTimestamp = number;
+
+export type ChartData = [UTCMilliTimestamp, number];
+export type PlotLine = [string, UTCMilliTimestamp];
+
+export const formatNumber = (value: number, decimals = 0): string => {
+  return Number(value).toLocaleString(undefined, {
+    minimumFractionDigits: decimals > 0 ? decimals : 0,
+    maximumFractionDigits: decimals > 0 ? decimals : 0,
+  });
+};
+
+export const formatRatingHistory = (
+  ratingHistory: HistoricalRating[],
+): Record<string, ChartData[]> | undefined => {
+  const ratingRecord: Record<string, ChartData[]> = {};
+  return ratingHistory.reduce((record, teamData) => {
+    if (teamData.team_rating !== undefined) {
+      record[teamData.team_rating.team.name] =
+        teamData.team_rating.rating_history.map((match) => [
+          match.timestamp.getTime(),
+          match.rating,
+        ]);
+    }
+    return record;
+  }, ratingRecord);
+};
+
```

Comment:
when I was doing the historical rating endpoint I realized the backend serializer serializes some extra information(remember serializing whole team object) seeing this formatRatingHistory game the idea of changing the backend exported format directly provide something a long the line of ChartData here
let me know what you think, could be an idea for later tho

Reply:
hmm yeah maybe later... I think that memoizing this transform in the frontend is ok for now :+1: